logstreams/src/main/java/io/zeebe/logstreams/impl/log/index/LogBlockIndex.java

```diff
@@ -42,222 +41,163 @@
-   *     position
-   */
-  protected int lookupIndex(long position) {
-    final int lastEntryIdx = size() - 1;
-
-    int low = 0;
-    int high = lastEntryIdx;
-
-    int idx = -1;
-
-    if (low == high) {
-      final int entryOffset = entryOffset(low);
-      final long entryValue = indexBuffer.getLong(entryLogPositionOffset(entryOffset));
-
-      if (entryValue <= position) {
-        idx = low;
-      }
-
-      high = -1;
-    }
-
-    while (low <= high) {
-      final int mid = (low + high) >>> 1;
-      final int entryOffset = entryOffset(mid);
-
-      if (mid == lastEntryIdx) {
-        idx = mid;
-        break;
-      } else {
-        final long entryValue = indexBuffer.getLong(entryLogPositionOffset(entryOffset));
-        final long nextEntryValue =
-            indexBuffer.getLong(entryLogPositionOffset(entryOffset(mid + 1)));
-
-        if (entryValue <= position && position < nextEntryValue) {
-          idx = mid;
-          break;
-        } else if (entryValue < position) {
-          low = mid + 1;
-        } else if (entryValue > position) {
-          high = mid - 1;
-        }
-      }
-    }
+          if (currentBlockPosition <= entryPosition) {
+            blockPosition.set(currentBlockPosition);
+            return true;
+          } else {
+            return false;
+          }
+        });
 
-    return idx;
+    return blockPosition.get();
   }
 
-  /**
-   * Invoked by the log Appender thread after it has first written one or more entries to a block.
-   *
-   * @param logPosition the virtual position of the block (equal or smaller to the v position of the
-   *     first entry in the block)
-   * @param storageAddr the physical address of the block in the underlying storage
-   * @return the new size of the index.
-   */
-  public int addBlock(long logPosition, long storageAddr) {
-    final int currentIndexSize =
-        indexBuffer.getInt(indexSizeOffset()); // volatile get not necessary
-    final int entryOffset = entryOffset(currentIndexSize);
-    final int newIndexSize = 1 + currentIndexSize;
-
-    if (newIndexSize > capacity) {
-      expandIndexBuffer();
-    }
-
-    if (lastVirtualPosition >= logPosition) {
+  public synchronized void addBlock(long blockPosition, long blockAddress) {
+    if (lastVirtualPosition >= blockPosition) {
       final String errorMessage =
           String.format(
               "Illegal value for position.Value=%d, last value in index=%d. Must provide positions in ascending order.",
-              logPosition, lastVirtualPosition);
+              blockPosition, lastVirtualPosition);
       throw new IllegalArgumentException(errorMessage);
     }
 
-    lastVirtualPosition = logPosition;
-
-    // write next entry
-    indexBuffer.putLong(entryLogPositionOffset(entryOffset), logPosition);
-    indexBuffer.putLong(entryAddressOffset(entryOffset), storageAddr);
+    lastVirtualPosition = blockPosition;
+    key.wrapLong(blockPosition);
```

Comment:
maybe we rename the key value fields to blockPosition and blockAddress?

Reply:
Sure. I had it like it initially but thought it could be confusing having the parameters and fields be named the same way.